app/nslc/models.py

```diff
@@ -178,13 +182,26 @@ def update_channels(self):
         # 3. Finish
         # Now actually add channels
         self.channels.set(channels)
+        self.save()
 
     def __str__(self):
         return self.name
 
-    def save(self, *args, **kwargs):
-        super().save(*args, **kwargs)
-        self.update_channels()
+
+def channels_changed(instance, action, **kwargs):
+
+    # after m2m change is saved
+    if action == 'post_add':
```

Comment:
would we want this to be triggered for post_remove and post_clear as well?

Reply:
Yes - good point